pkg/aws/awsapi/awsapi.go

```diff
@@ -0,0 +1,68 @@
+package aws
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/pricing"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+)
+
+type EC2API interface {
```

Comment:
calls to this package will stutter a bit with `awsapi.EC2API`. Maybe consider something like `sdk.EC2API` or similar. `awsapi.EC2` may also be an option